packages/core/src/App/Containers/Layout/app-contents.jsx

```diff
@@ -32,28 +31,34 @@ const AppContents = ({
     const [show_cookie_banner, setShowCookieBanner] = React.useState(false);
     const [is_gtm_tracking, setIsGtmTracking] = React.useState(false);
     const { is_appstore } = React.useContext(PlatformContext);
-    const {
-        client: { user_id },
-    } = useStore();
 
     const tracking_status = tracking_status_cookie.get(TRACKING_STATUS_KEY);
 
     const scroll_ref = React.useRef(null);
 
+    const current_page = window.location.hostname + window.location.pathname;
+
     React.useEffect(() => {
         // rudderstack page view trigger
-        WS.wait('authorize').then(() => {
-            RudderStack.identifyEvent(user_id, {
-                language: getLanguage().toLowerCase() || 'en',
-            });
-            const current_page = window.location.hostname + window.location.pathname;
-            RudderStack.pageView(current_page);
+        WS.wait('authorize').then(response => {
+            if (!response.error) {
+                const { user_id } = response.authorize;
+
+                if (is_logged_in && user_id) {
+                    RudderStack.identifyEvent(user_id, {
+                        language: getLanguage().toLowerCase() || 'en',
+                    });
+                    RudderStack.pageView(current_page);
+                }
+            }
```

Comment:
```suggestion
            if (response.error) return;
            const { user_id } = response.authorize;
            
            if (!is_logged_in && !user_id) return;

            RudderStack.identifyEvent(user_id, {
                language: getLanguage().toLowerCase() || 'en',
            });
            RudderStack.pageView(current_page);
```
More readable if you do an early return

Reply:
Honestly, I don't find it more readable, with the second change I feel like people might miss that the code is only for authorized users